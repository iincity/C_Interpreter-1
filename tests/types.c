#include <stdio.h>

double d = 2.48;


double sqrt(double x)
{
	double guess = 1.0;
	double tmp;
	while (1) {
		tmp = guess * guess - x;
		tmp = (tmp < 0) ? 0 - tmp : tmp;
		if (tmp < 0.001)
			return guess;
		guess = (guess + x/guess) / 2;
	}
}

int testfunc(int a, double b)
{
	printf("%d\n", a);
	printf("%f\n", b);
	return a + b;
}

short testshort(short a, unsigned short b)
{
	printf("%d\n", a);
	printf("%u\n", b);

	return a + b;
}


int main()
{
	short unsigned s = 100;
	int short unsigned crap = 123;
	int a = 5;

	printf("%f\n", d);
	printf("%d\n", a);

	a = d;
	d += a;
	printf("%f\n", d);
	printf("%d\n", a);

	d = sqrt(2.0);
	printf("%f\n", d);
	printf("%f\n", sqrt(25));

	printf("%d\n", testfunc(a, d));
	printf("%d\n", testfunc(d, d));

	d = 2.5;
	a = 70000;
	printf("%d\n", testfunc(a, d));
	printf("%d\n", testfunc(d, d));

	//2 4 8 so my conversions are ok here at least
	//printf("%lu %lu %lu\n", sizeof(short), sizeof(int), sizeof(long));
	return 0;
}
