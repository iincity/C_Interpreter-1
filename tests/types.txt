
double d = 2.48;

int main()
{
	int a = 5;
	print d;
	print a;

	a = d;
	d += a;
	print d;
	print a;

	return 0;
}
