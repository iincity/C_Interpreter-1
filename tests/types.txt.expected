2.480000
5
4.480000
2
1.414216
5.000023
2
1.414216
3
1
1.414216
2
70000
2.500000
70002
2
2.500000
4
-15536
50000
34464
